### ISSUE TITLE
When I want to access http service through TCP tunnel, the tunnel is often closed

### ISSUE THREAD

**Issue by chenjianhua** [rustdesk/rustdesk#487]
windows-->ubuntu.When I want to access http service through TCP tunnel, every time I open a webpage, the tunnel is closed. When I refresh the web page next time, I need to wait a long time to re-establish the tunnel. When the page is loaded, the tunnel is automatically closed. Can the tunnel remain open until I actively close it?

**Comment by rustdesk** [comment]
yes, this is a design problem, it is not friendly to short connection.

**Comment by rustdesk** [comment]
need a new design.

**Comment by rustdesk** [comment]
few guys use this feature.

**Comment by chenjianhua** [comment]
Software developers use this feature more than others. Remote office, open the http server on the controlled end, and the main control end can easily open the website and debug it. This is a very great mode.

**Comment by chenjianhua** [comment]
Looking forward to a better experience

**Comment by zy1818** [comment]
我也遇到这个问题，刚开始tcp隧道可以连，后面不知道什么原因再也连不上了

**Comment by 497983606** [comment]
> few guys use this feature.

i'm need my company many scene use i, plase getit, thank u

**Comment by cloudenum** [comment]
any update on this?

**Comment by Tools2** [comment]
> For me it works (very) stable already for a long time...

What version are you using? I often have issues with TCP tunnel forwarding when using this feature.

**Comment by mherrmann3** [comment]
I always used the most recent available version. It's still stable, **except** (I noticed that in the meantime): when setting up more than 3 port forwards, after a while, somehow none of them work anymore (they seem to be all forwarded to the first forwarded port).
**tl;dr**: when setting up only up to 3 port forwards, everything works fine.
